Mathlib/LinearAlgebra/ExteriorPower/Generators.lean

```diff
@@ -0,0 +1,381 @@
+  rw [Basis.coord_apply, Basis.repr_self_apply]
+  simp only [Finset.coe_orderIsoOfFin_apply, ne_eq] at hi
+  simp only [Finset.coe_orderIsoOfFin_apply, Ne.symm hi, ite_false]
+
+/-- If `b` is a basis of `M` (indexed by a linearly ordered type), then the family
+`ExteriorPower.ιMulti R n b` of the `n`-fold exterior products of its elements is linearly
+independent in the `n`th exterior power of `M`. -/
+lemma ιMulti_family_linearIndependent_ofBasis {I : Type*} [LinearOrder I] (b : Basis I R M) :
+    LinearIndependent R (ιMulti_family R n b) :=
+  linearIndependent_of_dualFamily _ (fun s => linearFormOfBasis R n b s.2)
+  (fun ⟨_, _⟩ ⟨_, _⟩ hst ↦
+  by simp only [ne_eq, Subtype.mk.injEq] at hst; simp only [Function.comp_apply]
+     exact linearFormOfBasis_apply_nondiag _ _ _ _ _ hst)
+  (fun ⟨_, _⟩ ↦ by simp only [Function.comp_apply]; apply linearFormOfBasis_apply_diag)
+
+/-- If `b` is a basis of `M` (indexed by a linearly ordered type), the basis of the `n`th
+exterior power of `M` formed by the `n`-fold exterior products of elements of `b`. -/
+noncomputable def BasisOfBasis {I : Type*} [LinearOrder I] (b : Basis I R M) :
+    Basis {s : Finset I // Finset.card s = n} R ((Λ[R]^n) M) :=
+  Basis.mk (v := ιMulti_family R n b) (ιMulti_family_linearIndependent_ofBasis _ _ _)
+  (by rw [span_top_of_span_top']; rw [Basis.span_eq])
+
+@[simp]
+lemma BasisOfBasis_coe {I : Type*} [LinearOrder I] (b : Basis I R M) :
+    DFunLike.coe (BasisOfBasis R n b) = ιMulti_family R n b := Basis.coe_mk _ _
+
+@[simp]
+lemma BasisOfBasis_apply {I : Type*} [LinearOrder I] (b : Basis I R M)
+    {s : Finset I} (hs : Finset.card s = n) :
+    BasisOfBasis R n b ⟨s, hs⟩ = ιMulti_family R n b ⟨s, hs⟩ := by rw [BasisOfBasis_coe]
+
+/-- If `b` is a basis of `M` indexed by a linearly ordered type `I` and `B` is the corresponding
+basis of the `n`th exterior power of `M`, indexed by the set of finsets `s` of `I` of cardinality
+`n`, then the coordinate function of `B` at `s` is the linear form on the `n`th exterior power
+defined by `b` and `s` in `ExteriorPower.linearFormOfBasis`. -/
+lemma BasisOfBasis_coord {I : Type*} [LinearOrder I] (b : Basis I R M)
+    {s : Finset I} (hs : Finset.card s = n) :
+    Basis.coord (BasisOfBasis R n b) ⟨s, hs⟩ = linearFormOfBasis R n b hs := by
+  apply LinearMap.ext_on (span_top_of_span_top' R n (Basis.span_eq b))
+  intro x hx
+  obtain ⟨⟨t, ht⟩, htx⟩ := Set.mem_range.mp hx
+  rw [← htx]
+  conv_lhs => rw [← BasisOfBasis_apply]
+  by_cases heq : s = t
+  · have heq' : (⟨s, hs⟩ : {s : Finset I // Finset.card s = n}) = ⟨t, ht⟩ := by
+      simp only [Subtype.mk.injEq]; exact heq
+    simp only [← heq', linearFormOfBasis_apply_diag, Basis.coord_apply, Basis.repr_self,
+      Finsupp.single_eq_same]
+  · simp only [linearFormOfBasis_apply_nondiag R n b hs ht heq, Basis.coord_apply,
+      Basis.repr_self_apply, Subtype.mk.injEq, Ne.symm heq, ite_false]
+
+/-! Freeness and dimension of `Λ[R]^n M. -/
```

Comment:
```suggestion
/-! ### Freeness and dimension of `Λ[R]^n M. -/
```
or
```suggestion
/-! #### Freeness and dimension of `Λ[R]^n M. -/
```
depending on whether you want it to be displayed as a subsubheader or a subsubsubheader in the docs.